tesseract-server/src/app.rs

```diff
@@ -56,14 +63,32 @@ pub fn create_app(backend: Box<dyn Backend + Sync + Send>, db_type: Database, sc
         .resource("/cubes/{cube}", |r| {
             r.method(Method::GET).with(metadata_handler)
         })
+
+        // Aggregation
         .resource("/cubes/{cube}/aggregate", |r| {
             r.method(Method::GET).with(aggregate_default_handler)
         })
         .resource("/cubes/{cube}/aggregate.{format}", |r| {
             r.method(Method::GET).with(aggregate_handler)
         })
+
+        // Aggregation + Logic Layer
```

Comment:
I apologize, I may have not communicated clearly enough about the api here.

My plan was to have a `/logic-layer` route, and no `/cubes/{cube}/logic-layer` route. This `logic-layer` route will handle any query that requires inference, such as the cube resolution by measure, dimension resolution by cut, and member/dim resolution for time.

I'll keep reading further to see exactly what differs between the logic layer route and the aggregate route, and may add more comments here.

Reply:
I assumed that we wanted to have the ability to specify which cube we want to query. One of the annoying things about the current logic layer, in my opinion, is that you can't do this. There are cases where two separate cubes have the same measure and dimension names. These cases were common in the Authority Health project. In those cases, it was impossible for the logic layer to differentiate between the different cubes so I had to change some names to things that don't make much sense. For example, I had to change a dimension name from `Sex` to `Diabetes Gender` so it could pick up that cube. This doesn't make much sense since gender is gender regardless of whether someone has diabetes. 

So personally I'd prefer to have a separate endpoint where the cube can be specified. As far as other logic layer functionality, everything after the cube detection step in the `/logic-layer` endpoint would be shared among both endpoints in the `finish_aggregation` method.

Reply:
Thanks so much for bringing up this point. Let's discuss offline a bit.